consoleApp/src/main/java/com/issoft/store/StoreApp.java

```diff
@@ -1,20 +1,69 @@
 package com.issoft.store;
+import org.xml.sax.SAXException;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.*;
+import java.util.*;
 
-import java.util.List;
 
 
 public class StoreApp {
 
 
-    public static void main(String args[]) throws IllegalAccessException {
+    public static void main(String args[]) throws IllegalAccessException, IOException, ParserConfigurationException, SAXException {
+
+
+        int i = 1;
+        while(i>=1){
+            System.out.println("Hi! Available commands: sort, top, exit. Please enter your command: ");
+            BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+            String userCommand = reader.readLine();
+            RandomStorePopulator randomStorePopulator = new RandomStorePopulator();
+            Store store = new Store();
+            store.setCategories(randomStorePopulator.populateProductLists());
+            List<Category> extractedData = randomStorePopulator.extractDataFromStore(store);
+            List<Product> extractedProducts = new ArrayList<>();
+            for (Category category: store.getCategories()) {
+                extractedProducts.addAll(category.getProducts());
+            }
+
+        if(userCommand.equalsIgnoreCase("sort")){
+
+            Comparator<Product> productComparator = new ProductNameComparator().thenComparing(new ProductRateComparator().thenComparing(new ProductPriceComparator()));
+            Collections.sort(extractedProducts, productComparator);
+            randomStorePopulator.prettyProducts(extractedProducts);
+            }
+        else if(userCommand.equalsIgnoreCase("top")){
+
+            ProductPriceComparator productPriceComparator = new ProductPriceComparator();
+            Collections.sort(extractedProducts, productPriceComparator);
+            extractedProducts.subList(5,7).clear();
```

Comment:
I suggest to save in new list first 5 products using subList(0,4) instead of cutting the last 3(right now).
It works now as you have 8 products. But what will happen if you have 10 products,
It will avoid situation when you change the number of products and need to update this string of code